pkg/api/validation/validation.go

```diff
@@ -169,6 +169,13 @@ func ValidateEndpointsName(name string, prefix bool) (bool, string) {
 	return nameIsDNSSubdomain(name, prefix)
 }
 
+// ValidateServiceName can be used to check whether the given component name is valid.
+// Prefix indicates this name will be used as part of generation, in which case
+// trailing dashes are allowed.
+func ValidateComponentName(name string, prefix bool) (bool, string) {
+	return nameIsDNS952Label(name, prefix)
```

Comment:
Most objects use nameIsDNSSubdomain() - why did you choose the much more restrictive DNS952Label?